fe/fe-core/src/main/java/com/starrocks/catalog/Database.java

```diff
@@ -620,7 +620,12 @@ public Pair<Table, MaterializedIndexMeta> getMaterializedViewIndex(String mvName
             if (table instanceof OlapTable) {
                 OlapTable olapTable = (OlapTable) table;
                 for (MaterializedIndexMeta mvMeta : olapTable.getVisibleIndexMetas()) {
-                    String indexName = olapTable.getIndexNameById(mvMeta.getIndexId());
+                    Long mvIndexId = mvMeta.getIndexId();
+                    // materialized view index should not consider the base index id.
+                    if (mvIndexId == olapTable.getBaseIndexId()) {
```

Comment:
why mvIndexId could equal to olapTable base index id ?

Reply:
this is used  to fix this issue: https://github.com/StarRocks/starrocks/issues/27110

`getMaterializedViewIndex ` should return only the single materialized view rather than the base table.